memory_report.cpp

```diff
@@ -143,25 +140,19 @@ void report_memory_consumption(
    MPI_Comm_rank(interComm, &interRank);
    MPI_Comm_size(interComm, &nNodes);
 
-#if __GNUC__ >= 13
-   // string formatting
-   constexpr std::string_view formatString = "(MEM) tstep {} t {:.3g} {:<21} (GiB/node; avg, min, max, sum): {:<8.3g} {:<8.3g} {:<8.3g} {:<8.3g} on {} nodes\n";
-#endif
-
    // Report /proc/meminfo memory consumption first so we then get resident and HWM from Papi below, easier to compare by eye in logfile in this order.
    double mem_proc_free = (double)get_node_free_memory();
    double total_mem_proc = 0;
    double min_free,max_free;
    const int numberOfParameters = 1;
-   MPI_Reduce( &mem_proc_free, &total_mem_proc, numberOfParameters, MPI_DOUBLE, MPI_SUM, MASTER_RANK, MPI_COMM_WORLD );
+   MPI_Reduce( &mem_proc_free, &total_mem_proc, numberOfParameters, MPI_DOUBLE, MPI_SUM, 0, interComm );
```

Comment:
Shouldn't it rather be a `nodeComm` communication from all tasks which have `interRank==0` or similar?

Reply:
up

Reply:
As far as I understand it: no.

The nodeComms are created on each node to contain *all ranks within that node*. So it only does local communication.
The interComm are created globally, and contain *the same ranks from each node*.

So we are now reducing memory information from all nodes, and then only printing from rank 0, which seems conceptually correct to me and also works as intended in my tests.

Reply:
(Compare the definition of [MPI_Comm_split](https://www.mpich.org/static/docs/v3.3/www3/MPI_Comm_split.html), which splits the communicator based on the ```color``` argument.):

```
MPI_Comm_split(MPI_COMM_WORLD, nodehash, rank, &nodeComm);
MPI_Comm_split(MPI_COMM_WORLD, nodeRank, rank, &interComm);
```